sn_transfers/src/wallet/hot_wallet.rs

```diff
@@ -484,11 +495,17 @@ impl HotWallet {
 
         let spend_reason = Default::default();
         let start = Instant::now();
+        let derivation_index = DerivationIndex::random(&mut rng);
         let offline_transfer = OfflineTransfer::new(
             available_cash_notes,
             recipients,
             self.address(),
             spend_reason,
+            Some((
+                self.key.main_pubkey(),
+                derivation_index,
+                self.key.derive_key(&derivation_index),
+            )),
         )?;
```

Comment:
Requiring the secret key `self.key` for the offline_transfer part of the code is a drawback of the middle spend. 
We can leave it for now as this goes out of scope of this PR. But I think we should keep that in mind as it makes offline tx preparation impossible without the key!